# **Oxygen limited condition**
## **flux balance analysis (FBA) with CobraPy (version 0.4.1) **

*Author: Yanfen Fu (fyf@uw.edu)*

### Four cases has been simulated based on bioreactor data for WT(FM86,FM87) and  ΔCyt aa3 (FM88, FM90)
    - case 1: WT without additional constraints, bounded with specific methane uptake rate from FM86 and FM87
    - case 2: WT with additional constraint of specific oxygen uptake rate from FM86 and FM87
    - case 3: WT with secreted products (formate and acetate) rate constraint
    - case 4:  ΔCyt aa3 mutant with specific uptake rate of oxygen and methane, with additional constraints of   secreted products.
    - case 5: ΔCyt aa3 with 13C result




#### 1.1 case 1a, specific methane uptake = 2.7 mmol/gCDW.hr from FM86  

In [2]:
# case 1a (FM86)
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation.xml')
model.optimize()
model.summary()
print model.reactions.EX_for_c.x

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -3.03              h2o_e      2.55               BIOMASS_M5GB1      0.040
ch4_e      -2.70              co2_c      1.13                                   
no3_e      -0.29              h_c        0.56                                   
pi_e       -0.05              ac_c       0.00                                   
so4_e      -0.01              glc_a_c    0.00                                   
mg2_e      -0.00                                                                
gal_bD_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                
0.0


#### 1.2 case 1b, specific methane uptake = 2.9 mmol/gCDW.hr from FM87

In [3]:
# case 1b (FM87)
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation.xml')
model.reactions.EX_ch4_e.lower_bound = -2.9
model.reactions.EX_ch4_e.upper_bound = -2.9
model.optimize()
model.summary()
print model.reactions.EX_for_c.x

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -3.26              h2o_e      2.74               BIOMASS_M5GB1      0.043
ch4_e      -2.90              co2_c      1.21                                   
no3_e      -0.31              h_c        0.60                                   
pi_e       -0.05              ac_c       0.00                                   
so4_e      -0.01              glc_a_c    0.00                                   
mg2_e      -0.00                                                                
gal_bD_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                
0.0


#### 2.1 case 2a, specific oxygen uptake = 2.9 mmol/gCDW.hr from FM86  

In [5]:
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation.xml')
model.reactions.EX_o2_e.lower_bound = -2.9
model.reactions.EX_o2_e.upper_bound = -2.9

model.optimize()
model.summary()

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -2.90              h2o_e      2.83               BIOMASS_M5GB1      0.034
ch4_e      -2.70              co2_c      0.88                                   
no3_e      -0.25              h_c        0.24                                   
pi_e       -0.04              ac_c       0.24                                   
so4_e      -0.01              glc_a_c    0.00                                   
mg2_e      -0.00                                                                
phe__l_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                


#### 2.2 case 2b, specific methane uptake = 3.1 mmol/gCDW.hr from FM87

In [6]:
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation.xml')
model.reactions.EX_ch4_e.lower_bound = -2.9
model.reactions.EX_ch4_e.upper_bound = -2.9
model.reactions.EX_o2_e.lower_bound = -3.1
model.reactions.EX_o2_e.upper_bound = -3.1

model.optimize()
model.summary()

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -3.10              h2o_e      3.07               BIOMASS_M5GB1      0.036
ch4_e      -2.90              co2_c      0.92                                   
no3_e      -0.27              ac_c       0.28                                   
pi_e       -0.04              h_c        0.22                                   
so4_e      -0.01              glc_a_c    0.00                                   
mg2_e      -0.00                                                                
met__l_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                


#### 3.1 case 3a, secreated products formate and acetate are constrained with experimental result (FM86) using python script create ratio.py, the input model '5GB1_base_fermentation_case3a.xml' -> Might need to trouble shoot '5GB1_base_fermentation_case3.xml' doesn't lead to an optimum result




In [8]:
# case 3a FM86
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation_case3a.xml')
model.reactions.EX_ac_c.lower_bound=0.001
model.reactions.EX_ac_c.upper_bound=0.002
model.reactions.EX_o2_e.lower_bound = -2.9
model.reactions.EX_o2_e.upper_bound = -2.9
model.optimize()
model.summary()

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -2.90              h2o_e      2.37               BIOMASS_M5GB1      0.033
ch4_e      -2.70              co2_c      1.40                                   
no3_e      -0.24              h2_c       1.12                                   
h_c        -0.13              ac_c       0.00                                   
pi_e       -0.04              glc_a_c    0.00                                   
so4_e      -0.00                                                                
mg2_e      -0.00                                                                
ile__l_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                
0.0156724377449


#### 3.2 case 3b, secreated products formate and acetate are constrained with experimental result (FM87) using python script create ratio.py, the input model '5GB1_base_fermentation_case3r2.xml


In [10]:
# case 3b FM87
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation_case3r2.xml')
model.reactions.EX_o2_e.lower_bound = -3.1
model.reactions.EX_o2_e.upper_bound = -3.1
model.reactions.EX_ch4_e.lower_bound = -2.9
model.reactions.EX_ch4_e.upper_bound = -2.9
model.optimize()
model.summary()
print model.reactions.EX_for_c.x
print model.reactions.EX_ac_c.x

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -3.10              h2o_e      2.53               BIOMASS_M5GB1      0.034
ch4_e      -2.90              co2_c      1.55                                   
no3_e      -0.25              h2_c       1.34                                   
h_c        -0.23              glc_a_c    0.00                                   
pi_e       -0.04                                                                
so4_e      -0.01                                                                
mg2_e      -0.00                                                                
glu__l_c   -0.00                                                                
ca2_e      -0.00                                                                
cu2_e      -0.00                                                                
fe2_e      -0.00                                                                
0.00696628272135
0.00133

#### 4.1 case 4a, ΔCyt aa3 mutant, secreated products formate and acetate are constrained with experimental result (FM88) using python script create ratio.py, the input model '5GB1_base_fermentation_aa3_case3.xml'


In [12]:
# case 4a. ΔCyt aa3 model with bioreactor data FM88
# FM88 run, Oxygen uptake rate = 3.9 mmol/ gCDW.hr Methane uptake rate = 3.6 mmol/gCDW.hr
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation_aa3_case3.xml')
## aa3 complex flux bounded to 0
model.reactions.CyAA3_CYTC.upper_bound = 0
model.reactions.CyAA3_CYTC.lower_bound = 0
model.reactions.CyAA3_UQ.upper_bound = 0
model.reactions.CyAA3_UQ.lower_bound = 0
model.reactions.EX_o2_e.lower_bound = -3.9
model.reactions.EX_o2_e.upper_bound = -3.9
model.reactions.EX_ch4_e.lower_bound = -3.6
model.reactions.EX_ch4_e.upper_bound = -3.6
model.optimize()
model.summary()


IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e       -3.90              h2o_e      3.21               BIOMASS_M5GB1      0.043
ch4_e      -3.60              co2_c      1.85                                   
no3_e      -0.31              h2_c       1.45                                   
h_c        -0.15              glc_a_c    0.00                                   
pi_e       -0.05                                                                
so4_e      -0.01                                                                
mg2_e      -0.00                                                                
ile__l_c   -0.00                                                                
cu2_e      -0.00                                                                
ca2_e      -0.00                                                                
fe2_e      -0.00                                                                


#### 4.2 case 4b, ΔCyt aa3 mutant, secreated products formate and acetate are constrained with experimental result (FM90) using python script create ratio.py, the input model '5GB1_base_fermentation_aa3_case3r2.xml'

In [15]:
# case 4a. ΔCyt aa3 model with bioreactor data FM88
# FM88 run, Oxygen uptake rate = 4.5 mmol/ gCDW.hr Methane uptake rate = 4.2 mmol/gCDW.hr
import cobra
model = cobra.io.read_sbml_model('fermentation_case_study/5GB1_base_fermentation_aa3_case3r2.xml')
## aa3 complex flux bounded to 0
model.reactions.CyAA3_CYTC.upper_bound = 0
model.reactions.CyAA3_CYTC.lower_bound = 0
model.reactions.CyAA3_UQ.upper_bound = 0
model.reactions.CyAA3_UQ.lower_bound = 0
model.reactions.EX_o2_e.lower_bound = -4.5
model.reactions.EX_o2_e.upper_bound = -4.5
model.reactions.EX_ch4_e.lower_bound = -4.2
model.reactions.EX_ch4_e.upper_bound = -4.2
model.optimize()
model.summary()

IN FLUXES                     OUT FLUXES                    OBJECTIVES          
o2_e      -4.50               h2o_e      3.61               BIOMASS_M5GB1      0.047
ch4_e     -4.20               co2_c      2.20                                   
no3_e     -0.34               h2_c       2.08                                   
h_c       -0.29               glc_a_c    0.00                                   
pi_e      -0.05                                                                 
so4_e     -0.01                                                                 
mg2_e     -0.00                                                                 
glc_a_c   -0.00                                                                 
ca2_e     -0.00                                                                 
cu2_e     -0.00                                                                 
fe2_e     -0.00                                                                 


### The following table summarized case 1, 2, 3,4 simulation results
| Cases | Simulated growth rate | Strain |
|----- -|-----------------------|--------|
|   1a  | 0.04                  |   WT   | 
|   1b  | 0.043                 |   WT   |
|   2a  | 0.034                 |   WT   |
|   2b  | 0.036                 |   WT   |
|   3a  | 0.033                 |   WT   |
|   3b  | 0.034                 |   WT   |
|   4a  | 0.043                 |ΔCyt aa3   |
|   4b  | 0.047                 |ΔCyt aa3   |

